app/Http/Controllers/BaseHomeController.php

```diff
@@ -203,4 +204,87 @@ public function updateLatestVersion(Request $request)
+            if ($order->serial_key == $licCode) {//Check in the orders table and get the row that matches with the license code sent in the request
+                    return $order;
+                }
+            });
+            if (count($existingLicense) > 0) {//If the license code that is sent in the request exists in billing 
+                $cont = new \App\Http\Controllers\License\LicenseController();
+                $cont->updateInstalledDomain($licCode,$existingLicense->first()->product);//Delete the installation first for the current license before updating license so that no Faveo installation exists on the user domain/IP path 
+                $orderId = $existingLicense->first()->id;
+                $user_id = $existingLicense->first()->client;
+                $product = $existingLicense->first()->product;
+
+                $updatesExpiry = $this->getUpdatesExpiryDate($existingLicense);
+                $licenseExpiry = $this->getLicenseExpiryDate($existingLicense);
+                $supportExpiry = $this->getSupportExpiryDate($existingLicense);
+                
+                $serial_key = substr($licCode, 0,12).$lastFour;//The new License Code
+                //Create new license in license manager with the new license code which has no. of agents in the last 4 digits.
+                $cont->createNewLicene($orderId, $product, $user_id, $licenseExpiry, $updatesExpiry, $supportExpiry, $serial_key);
+                //Update the old license code with new one in billing. 
+                $existingLicense->first()->serial_key = \Crypt::encrypt(substr($licCode, 0,12).$lastFour);
+                $existingLicense->first()->save();
+                //send the newly updated license code in response
+                $result = ['status'=>'success', 'updatedLicenseCode'=>$existingLicense->first()->serial_key];
+                return response()->json($result);
+            }
+        } catch (\Exception $ex) {
+            $result = ['status'=>'fails', 'error' => $ex->getMessage()];
+            return response()->json($result);
+        }
+    }
+
+    public function getLastFourDigistsOfLicenseCode($productName)
+    {
+        if(strpos($productName, 'Enterprise') || strpos($productName, 'Company')) {
+            $digits =  '0000';//Unlimited Agents
+        } elseif(strpos($productName, 'Freelancer')) {
+            $digits =  '0002'; //2 Agents
+        } elseif(strpos($productName, 'Startup')) {
+            $digits = '0005'; //5 Agents
+        } elseif(strpos($productName, 'SME')) {
+            $digits =  '0010';
+        } else {
+             throw new \Exception(\Lang::get('message.product_not_found'));
+        }
+    
+        return $digits;
+
+    }
+
+    public function getUpdatesExpiryDate($existingLicense)
+    {
+       $updatesDate  = \Carbon\Carbon::parse(Subscription::where('order_id', $existingLicense->first()->id)->pluck('update_ends_at')->first());
```

Comment:
instead of pluck and then first, you could have used `->value('update_ends_at')`